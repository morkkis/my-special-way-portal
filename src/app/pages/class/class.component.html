<div class="main-content">
  <div class="container-fluid">
    <div class="row">
      <div class="example-header">
        <mat-form-field floatPlaceholder="never" color="accent">
          <input matInput #filter placeholder="Filter issues">
        </mat-form-field>
      </div>
      <div class="example-header">
        <button mat-raised-button color="primary">הוספת כיתה חדש</button>
      </div>
    </div>
    <div class="row">

      <div class="col-lg-12 col-md-12 col-sm-12">
        <mat-table class="example-table" matSort matSortActive="created" matSortDisableClear matSortDirection="asc">

          <!--- Note that these columns can be defined in any order.
                The actual rendered columns are set as a property on the row definition" -->

          <!-- Number Column -->
          <ng-container matColumnDef="name">
            <mat-header-cell *matHeaderCellDef>שם הכיתה</mat-header-cell>
            <mat-cell *matCellDef="let row">{{ row.name }}</mat-cell>
          </ng-container>

          <!-- Title Column -->
          <ng-container matColumnDef="level">
            <mat-header-cell *matHeaderCellDef>כיתה</mat-header-cell>
            <mat-cell *matCellDef="let row">{{ row.level }}</mat-cell>
          </ng-container>

          <!-- State Column -->
          <ng-container matColumnDef="teacherName">
            <mat-header-cell *matHeaderCellDef>שם המורה</mat-header-cell>
            <mat-cell *matCellDef="let row">{{ row.level }}</mat-cell>
          </ng-container>

          <!--&lt;!&ndash; State Column &ndash;&gt;-->
          <ng-container matColumnDef="editDetails">
            <mat-header-cell *matHeaderCellDef>עריכת פרטים</mat-header-cell>
            <mat-cell *matCellDef="let row">
              <button mat-icon-button color="primary" (click)="editClass()">ערוך</button>
            </mat-cell>
          </ng-container>

          <!--&lt;!&ndash; State Column &ndash;&gt;-->
          <ng-container matColumnDef="deleteClass">-->
            <mat-header-cell *matHeaderCellDef>מחיקת כיתה</mat-header-cell>
            <mat-cell *matCellDef="let row">
              <button mat-icon-button color="primary" (click)="deleteClass()">מחק</button>
            </mat-cell>
          </ng-container>

          <mat-header-row></mat-header-row>
          <mat-row></mat-row>
        </mat-table>

        <mat-paginator #paginator [length]="1" [pageIndex]="0" [pageSize]="10" [pageSizeOptions]="[5, 10, 25, 100]">
        </mat-paginator>
      </div>
    </div>
  </div>
</div>


<!-- Copyright 2018 Google Inc. All Rights Reserved.
    Use of this source code is governed by an MIT-style license that
    can be found in the LICENSE file at http://angular.io/license -->
