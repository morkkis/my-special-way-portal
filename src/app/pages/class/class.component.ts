import {Component} from '@angular/core';
import 'rxjs/add/observable/merge';
import 'rxjs/add/observable/fromEvent';
import 'rxjs/add/operator/map';
import 'rxjs/add/operator/debounceTime';
import 'rxjs/add/operator/distinctUntilChanged';

@Component({
  selector: 'app-grade',
  templateUrl: './class.component.html',
  styleUrls: ['./class.component.scss'],
})
export class ClassComponent {

  constructor() {}
}
