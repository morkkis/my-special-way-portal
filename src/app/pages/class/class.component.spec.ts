import { ClassComponent } from './class.component';
import { TestBed } from '@angular/core/testing';
import { NO_ERRORS_SCHEMA } from '@angular/core';

describe('dashboard component', () => {
  beforeEach(async () => {
    TestBed.configureTestingModule({
      declarations: [
        ClassComponent,
      ],
      imports: [],
      schemas: [NO_ERRORS_SCHEMA],
    });
  });

  it('should render component as described in snapshot', () => {
    const fixture = TestBed.createComponent(ClassComponent);
    expect(fixture).toMatchSnapshot();
  });
});
