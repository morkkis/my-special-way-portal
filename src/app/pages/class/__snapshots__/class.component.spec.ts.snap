// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`dashboard component should render component as described in snapshot 1`] = `
<app-grade>
  <div
    class="main-content"
  >
    
    
    <div
      class="container-fluid"
    >
      
      
      <div
        class="row"
      >
        
        
        <div
          class="example-header"
        >
          
          
          <mat-form-field
            color="accent"
            floatplaceholder="never"
          >
            
            
            <input
              matinput=""
              placeholder="Filter issues"
            />
            
          
          </mat-form-field>
          
        
        </div>
        
        
        <div
          class="example-header"
        >
          
          
          <button
            color="primary"
            mat-raised-button=""
          >
            הוספת כיתה חדש
          </button>
          
        
        </div>
        
      
      </div>
      
      
      <div
        class="row"
      >
        
  
        
        <div
          class="col-lg-12 col-md-12 col-sm-12"
        >
          
          
          <mat-table
            class="example-table"
            matsort=""
            matsortactive="created"
            matsortdirection="asc"
            matsortdisableclear=""
          >
            
  
            
            
  
            
            
            
            
            
              
            
            
              
            
            
            
            
  
            
            
            
            
            
              
            
            
              
            
            
            
            
  
            
            
            
            
            
              
            
            
              
            
            
            
            
  
            
            
            
            
            
              
            
            
              
            
            
            
            
  
            
            
            
            
            --&gt;
              
            
            
              
            
            
            
            
  
            
            <mat-header-row />
            
            
            <mat-row />
            
          
          </mat-table>
          
  
          
          <mat-paginator>
            
          
          </mat-paginator>
          
        
        </div>
        
      
      </div>
      
    
    </div>
    
  
  </div>
  
  
  
  
</app-grade>
`;
